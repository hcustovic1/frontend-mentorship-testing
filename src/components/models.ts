export interface ListItemType {
  title: string;
  description?: string;
}
