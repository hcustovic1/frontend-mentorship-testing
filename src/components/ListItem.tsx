import { ListItemType } from './models';

export interface ListItemProps {
  item: ListItemType;
}

const ListItem = ({ item: { title, description } }: ListItemProps) => {
  return (
    <article>
      <header>
        <h1>{title}</h1>
      </header>
      <div>
        <p>{description}</p>
      </div>
    </article>
  );
};

export default ListItem;
