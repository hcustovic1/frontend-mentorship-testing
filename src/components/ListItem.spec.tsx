import { render } from '@testing-library/react';
import ListItem, { ListItemProps } from './ListItem';
import { ListItemType } from './models';

const renderComponent = ({ item: { title, description } }: ListItemProps) => {
  return render(<ListItem item={{ title, description }} />);
};

describe('ListItem.tsx', () => {
  const listItemMock: ListItemType = {
    title: 'TITLE MOCK2',
    description: 'DESCRIPTION MOCK'
  };
  it('renders the component successfully', () => {
    const { container } = renderComponent({ item: listItemMock });

    expect(container.firstChild).toMatchSnapshot();
  });
});
