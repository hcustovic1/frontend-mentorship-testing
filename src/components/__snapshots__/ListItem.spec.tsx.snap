// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem.tsx renders the component successfully 1`] = `
<article>
  <header>
    <h1>
      TITLE MOCK2
    </h1>
  </header>
  <div>
    <p>
      DESCRIPTION MOCK
    </p>
  </div>
</article>
`;
